autoload/maktaba/syscall.vim

```diff
@@ -67,6 +69,24 @@ function! s:DoSyscallCommon(syscall, CallFunc, throw_errors) abort
 endfunction
 
 
+" Execute a function in given tab, used for callbacks of asynchronous actions.
+function! s:CallbackInTab(callback, source_tab) abort
+  call maktaba#ensure#IsCallable(a:callback)
+  call maktaba#ensure#IsNumber(a:source_tab)
+  let l:current_tab = tabpagenr()
+  let l:state = maktaba#value#Save('lazyredraw')
+  let &lazyredraw = 1
+  try
+    silent! execute "tabnext " . a:source_tab
+    call maktaba#function#Apply(a:callback)
+    silent! execute "tabnext " . l:current_tab
+    redraw!
```

Comment:
`redraw!` will wipe existing output like error messages from the callback. I think that will make it really hard to debug. Any reason we can't use `redraw` without the bang?